src/orion/core/utils/tests.py

```diff
@@ -0,0 +1,180 @@
+# -*- coding: utf-8 -*-
+"""
+:mod:`orion.core.utils.state` -- Encapsulate Orion state
+=================================================================
+.. module:: state
+   :platform: Unix
+   :synopsis: Encapsulate orion state
+
+"""
+
+import datetime
+import os
+import tempfile
+
+import yaml
+
+from orion.core.io.database import Database
+from orion.core.io.database.ephemeraldb import EphemeralDB
+from orion.core.io.database.mongodb import MongoDB
+from orion.core.io.database.pickleddb import PickledDB
+from orion.core.utils import SingletonAlreadyInstantiatedError
+from orion.core.worker.experiment import Experiment
+from orion.core.worker.trial import Trial
+from orion.storage.base import get_storage, Storage
+from orion.storage.legacy import Legacy
+
+
+def _remove(file_name):
+    try:
+        os.remove(file_name)
+    except FileNotFoundError:
+        pass
+
+
+def _select(lhs, rhs):
+    if lhs:
+        return lhs
+    return rhs
+
+
+def default_datetime():
+    """Return default datetime"""
+    return datetime.datetime(1903, 4, 25, 0, 0, 0)
+
+
+class MockDatetime(datetime.datetime):
+    """Fake Datetime"""
+
+    @classmethod
+    def utcnow(cls):
+        """Return our random/fixed datetime"""
+        return default_datetime()
+
+
+# pylint: disable=no-self-use,protected-access
+class OrionState:
+    """Setup global variables and singleton for tests
+
+    it swaps the singleton with none at startup and restore them after the tests.
+    It also initialize PickleDB as the storage for testing.
+    We use PickledDB as our storage mock
+
+    Parameters
+    ----------
+    config: YAML
+        YAML config to apply for this test
+
+    Examples
+    --------
+    >>> myconfig = {...}
+    >>> with OrionTestState(myconfig):
+        ...
+
+    """
+
+    SINGLETONS = (Storage, Legacy, Database, MongoDB, PickledDB, EphemeralDB)
```

Comment:
Maybe add a TODO comment: Fix these singletons to remove Legacy, MongoDB, PickledDB and EphemeralDB.